marin/run/slurm_run.py

```diff
@@ -0,0 +1,265 @@
+import argparse
+import logging
+import os
+import re
+import shlex
+import subprocess
+import sys
+import tempfile
+import time
+
+from huggingface_hub import HfFolder
+
+import wandb
+
+# Setup logger
+logger = logging.getLogger("slurm")
+
+# Default configuration
+DEFAULT_ENV_VARS = {
+    "PYTHONUNBUFFERED": "1",
+}
+
+DEFAULT_PIP_DEPS = []
+
+# SLURM specific defaults
+DEFAULT_SLURM_ARGS = {
+    "job-name": "marin_run",
+    "output": "logs/marin-%j.out",
+    "error": "logs/marin-%j.err",
+    "time": "48:00:00",
+    "mem": "200G",
+    "gres": "gpu:h200:1",
+    "account": "nlp",
+    "partition": "sc-loprio",
+    "constraint": "[40G|48G|80G|141G]",
+    "nodes": "1",
+    "ntasks-per-node": "1",
+    "cpus-per-task": "64",
+}
+
+
+def parse_pip_requirements(line: str) -> list[str]:
+    pattern = r"(?:\[[^\]]*\]|[^,])+"
+    return re.findall(pattern, line)
+
+
+def generate_pythonpath(base_dir="submodules"):
+    paths = []
+    if not os.path.exists(base_dir):
+        logger.warning(f"Base directory {base_dir} does not exist.")
+        return ""
+    for submodule in os.listdir(base_dir):
+        submodule_path = os.path.join(base_dir, submodule)
+        if os.path.isdir(submodule_path):
+            paths.append(submodule_path)
+            src_path = os.path.join(submodule_path, "src")
+            if os.path.isdir(src_path):
+                paths.append(src_path)
+    return ":".join(paths)
+
+
+def create_sbatch_script(
```

Comment:
Is it better to use Python's slurm package?

Reply:
Which package do you mean? I used https://github.com/facebookincubator/submitit for a while at Meta, which I think is the most used package but not sure if you mean a different one.

Regardless, I feel like it's better to avoid having an extra dependency as long as the logic remains pretty simple.

Reply:
Yeah, submitit came to mind.  If we don't like the abstraction, it's fine to not use it.  But it seems like a pretty lightweight dependency compare to other things we have.